packages/vite/src/node/plugins/clientInjections.ts

```diff
@@ -66,7 +66,7 @@ export function clientInjectionsPlugin(config: ResolvedConfig): Plugin {
         // for it to avoid shimming a `process` object during dev,
         // avoiding inconsistencies between dev and build
         return code.replace(
-          /\bprocess\.env\.NODE_ENV\b/g,
+          /(\bglobalThis\.)?\bprocess\.env\.NODE_ENV\b/g,
```

Comment:
Looks like we also support `global`:

https://github.com/vitejs/vite/blob/3d3788d5518350dd989659548ce0610134781851/packages/vite/src/node/plugins/define.ts#L27-L29

We could probably tweak the regex to support it too.

Reply:
done